numhask-test/src/NumHask/Laws.hs

```diff
@@ -119,262 +121,359 @@ testLawOf2 _ (name, Quad31 f) = testProperty name f
-  [Law2 a b]
+metricIntegralLaws
+  :: forall a b
+   . ( FromInteger b
+     , Ord b
+     , Signed b
+     , Epsilon b
+     , Metric a b
+     , AbelianGroup (Sum b)
+     )
+  => [Law2 a b]
 metricIntegralLaws =
-  [ ("Lp: positive",
-     Ternary21 (\a b p -> p < one || distanceLp p a b >= zero))
-  , ("Lp: zero if equal"
-    , Binary11 (\a p -> p < one || distanceLp p a a == zero))
+  [ ("Lp: positive", Ternary21 (\a b p -> p < one || distanceLp p a b >= zero))
+  , ( "Lp: zero if equal"
+    , Binary11 (\a p -> p < one || distanceLp p a a == zero)
+    )
   , ( "Lp: associative"
-    , Ternary21 (\a b p ->
-                  p < one ||
-                  p > (smallIntegralPower :: b) ||
-                 distanceLp p a b ~= distanceLp p b a))
+    , Ternary21
+      (\a b p ->
+        p
+          < one
+          || p
+          > (smallIntegralPower :: b)
+          || distanceLp p a b
+          ~= distanceLp p b a
+      )
+    )
   , ( "Lp: triangle rule - sum of distances > distance"
     , Quad31
-        (\a b c p ->
-           (p < one) ||
-           not
-             (veryNegative
-                (distanceLp p a c + distanceLp p b c - distanceLp p a b)) &&
-           not
-             (veryNegative
-                (distanceLp p a b + distanceLp p b c - distanceLp p a c)) &&
-           not
-             (veryNegative
-                (distanceLp p a b + distanceLp p a c - distanceLp p b c))))
+      (\a b c p ->
+        (p < one)
+          || not
+              (veryNegative
+                (distanceLp p a c + distanceLp p b c - distanceLp p a b)
+              )
+          && not
+              (veryNegative
+                (distanceLp p a b + distanceLp p b c - distanceLp p a c)
+              )
+          && not
+              (veryNegative
+                (distanceLp p a b + distanceLp p a c - distanceLp p b c)
+              )
+      )
+    )
   ]
 
 -- triangle rule doesn't apply to bounded Integrals
-metricIntegralBoundedLaws :: forall a b. (FromInteger b, Bounded b, Ord b, Signed b, Epsilon b, Metric a b) =>
-  [Law2 a b]
+metricIntegralBoundedLaws
+  :: forall a b
+   . (FromInteger b, Bounded b, Ord b, Signed b, Epsilon b, Metric a b)
+  => [Law2 a b]
 metricIntegralBoundedLaws =
-  [ ("Lp: positive",
-     Ternary21 (\a b p -> p < one || distanceLp p a b >= zero || distanceLp p a b == minBound))
-  , ("Lp: zero if equal"
-    , Binary11 (\a p -> p < one || distanceLp p a a == zero))
+  [ ( "Lp: positive"
+    , Ternary21
+      (\a b p ->
+        p < one || distanceLp p a b >= zero || distanceLp p a b == minBound
+      )
+    )
+  , ( "Lp: zero if equal"
+    , Binary11 (\a p -> p < one || distanceLp p a a == zero)
+    )
   , ( "Lp: associative"
-    , Ternary21 (\a b p ->
-                  p < one ||
-                  p > (smallIntegralPower :: b) ||
-                 distanceLp p a b ~= distanceLp p b a))
+    , Ternary21
+      (\a b p ->
+        p
+          < one
```

Comment:
I find all these linebreaks make the formula harder to read than before the PR

Reply:
How to minimise readability - it's almost art.  I agree with @LeanderK that its better to accept a dumb styler than have to edit by hand.  #99 addresses the next few comments - I found brackets help brittany out.